drivers/storage/portworx/portworx_test.go

```diff
@@ -202,11 +203,56 @@ func TestGetStorkEnvMap(t *testing.T) {
 	require.Equal(t, "true", envVars[pxutil.EnvKeyPortworxEnableTLS].Value)
 }
 
+func TestSetDefaultsOnStorageClusterWithVersionConfigMap(t *testing.T) {
+	namespace := "kube-test"
+	versionsConfigMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      manifest.DefaultConfigMapName,
+			Namespace: namespace,
+		},
+		Data: map[string]string{
+			manifest.VersionConfigMapKey: `
+version: 3.2.1
+components:
+  stork: stork/image:3.2.1
+`,
+		},
+	}
+
+	k8sClient := testutil.FakeK8sClient(versionsConfigMap)
+	coreops.SetInstance(coreops.New(fakek8sclient.NewSimpleClientset()))
+	driver := portworx{}
+	recorder := record.NewFakeRecorder(100)
+	err := driver.Init(k8sClient, runtime.NewScheme(), recorder)
+	require.NoError(t, err)
+	cluster := &corev1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: namespace,
+		},
+		Spec: corev1.StorageClusterSpec{
+			Image: "test/image:2.10.0",
```

Comment:
hmm still confused, in this case shouldn't the version here overwrite the one in the configmap?

Reply:
no it should not, we need to use same version in storagecluster and px-versions configmap, there is complaint that operator does not right version, and if the versions do not match there is no source of truth. in px-versions configmap scenario, to update px: 1. update px-versions configmap. 2. update storageCluster.spec.image.

Reply:
ok so we should add one more validation here, like the desired stork image should not be updated in this case and should not be empty.